### DISCUSSION TITLE
How do you test aggregate-partials?

### DISCUSSION THREAD

**Discussion by ovp87** [spatie/laravel-event-sourcing#404] [Q&A] [answered]
Hi, thanks for the work on an awesome package!

I must be missing something, i can't seem to wrap my head around a nice way to test aggregatePartials behaviour.
My immediate thought was something like this:

```
$fakeRoot = SomeAggregateRoot::fake()->given([
     new SomeOtherEvent()
]);

$aggregatePartial = ImplementedAggregatePartial::fake($fakeRoot);
$aggregatePartial->doSomething();
$aggregatePartial->assertRecorded([
    new SomeEvent(x, $y, $z)
])
```

But obviously this is not how it's meant to be used. Can anyone point me in the right direction?
Is the idea that you should only ever really test the AggregateRoot?

**Comment by Shkeats** [comment]
Hi @ovp87 I think the approach you've listed is actually the right one. You can depend on the test for the AggregateRoot itself to test the behaviour or cover it directly by using fake() on the partial.

If you want to learn more its actually covered specifically in the Spatie Event Sourcing book on pages 135 & 136.

**Comment by sebastiandedeyne** [comment, marked as answer]
Aggregate partials are like traits: they're a nice way to split a class in multiple parts but still should be considered part of that class.

Ideally, the aggregate root is the subject of your test. Partials are an implementation detail.
